shared/src/pool_aggregating.rs

```diff
@@ -59,15 +60,19 @@ impl PoolAggregator {
 
 #[async_trait::async_trait]
 impl PoolFetching for PoolAggregator {
-    async fn fetch(&self, token_pairs: HashSet<TokenPair>, at_block: BlockNumber) -> Vec<Pool> {
-        futures::future::join_all(
+    async fn fetch(
+        &self,
+        token_pairs: HashSet<TokenPair>,
+        at_block: BlockNumber,
+    ) -> Result<Vec<Pool>> {
+        // vk: Using try join means if any pool fetcher fails we fail too. Alternatively we could
+        // return the succeeding ones but I feel it is cleaner to forward the error.
```

Comment:
I guess this boils down to seeing how reliable the service will be with this configuration.